pkg/acsengine/transform.go

```diff
@@ -111,7 +114,7 @@ func (t *Transformer) NormalizeForK8sVMASScalingUp(logger *logrus.Entry, templat
 
 		for dIndex := len(dependencies) - 1; dIndex >= 0; dIndex-- {
 			dependency := dependencies[dIndex].(string)
-			if strings.Contains(dependency, nsgResourceType) {
+			if strings.Contains(dependency, nsgResourceType) || strings.Contains(dependency, nsgID) {
```

Comment:
I added one. It took a lot of work as when this code was moved from the service code to acs-engine the unit tests weren't fully brought over.